app/Http/Controllers/Admin/BookController.php

```diff
@@ -24,6 +27,52 @@ public function create()
         return view('backend.books.create', compact('categories'));
     }
 
+    /**
+     * Store a newly book created resource in storage.
+     *
+     * @param App\Http\Requests\BookCreateRequest $request get create request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(BookCreateRequest $request)
+    {
+        $book = new Book($request->toArray());
+        // save image path, move image to directory
+        if (isset($request->image)) {
+            $image = $request->image;
+            $name = config('image.name_prefix') . "-" . $image->hashName();
+            $folder = config('image.books.path_upload');
+            $image->move($folder, $name);
+
+            $path = $folder . $name;
+            $book->image = $path;
+        } else {
+            $path = config('image.no_image');
+            $book->image = $path;
+        }
+        //save generate qrcode
+        $book->QRcode = $book->generateQRcode();
+
+        //save new donator
+        $user = User::where('employee_code', $request->employee_code)->first();
+        $donatorData = [
+            'user_id' => $user->id,
+            'employee_code' => $user->employee_code,
+            'email' => $user->email,
+        ];
+        $donator = Donator::firstOrCreate($donatorData);
```

Comment:
I think u should edit updateOrCreate($donatorData). We can update donator after time